src/core/UserIconGroup/UserIconGroup.tsx

```diff
@@ -0,0 +1,148 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { useTheme } from '../utils/hooks/useTheme';
+import cx from 'classnames';
+import { CommonProps } from '../utils/props';
+
+export type UserIconGroupProps = {
+  /**
+   * Max number of icons unstacked.
+   * @default 5
+   */
+  maxIcons?: number;
+  /**
+   * If true, group will be stacked to take less space.
+   * @default true
+   */
+  stacked?: boolean;
+  /**
+   * If true, group will be animated on hover.
+   * @default false
+   */
+  animated?: boolean;
+  /**
+   *  Size of user and count icons.
+   */
+  iconSize: 'small' | 'medium' | 'large' | 'x-large';
+  /**
+   * User Icons in the group.
+   */
+  children: React.ReactNode;
+  /**
+   * User Count Icon props.
+   */
+  countIconProps?: React.HTMLAttributes<HTMLDivElement>;
+} & Omit<CommonProps, 'title'>;
+/**
+ * Group User Icons together.
+ *
+ * User Icons stacking is based on `maxIcons` count. If you provide 8 User Icons and keep default 5 `maxIcons` count,
+ * this component will show 5 User Icons and User Count Icon with "3" in it.
+ *
+ * You can add custom User Count Icon behavior by using `countIconProps`.
+ *
+ * @example
+ * <UserIconGroup iconSize='medium'>
+ *  <UserIcon
+ *    abbreviation="TR"
+ *    backgroundColor="#6AB9EC"
+ *    size="medium"
+ *    title="Terry Rivers"
+ *    type="default"
+ *  />
+ *  <UserIcon
+ *   abbreviation="RM"
+ *    backgroundColor="#C8C2B4"
+ *    size="medium"
+ *    title="Robin Mercer"
+ *    type="default"
+ *  />
+ *  <UserIcon
+ *    abbreviation="TR"
+ *    backgroundColor="#6AB9EC"
+ *    size="medium"
+ *    title="Terry Rivers"
+ *    type="default"
+ *  />
```

Comment:
Update this example.

1. Names should probably not be repeated.
2. Background color should use `getUserColor` function.
3. Size does not need be set on on user icons.
4. There is no `type` prop.